future-apps/payroll/contracts/Payroll.sol

```diff
@@ -793,21 +789,24 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
         uint256 accruedSalary = employee.accruedSalary;
 
         if (_paymentAmount <= accruedSalary) {
-            // Employee is only cashing out some previous owed salary, so we don't need to update the last payroll date
-            // No need to use SafeMath here since we already know that _paymentAmount <= accruedSalary
+            // Employee is only cashing out some previously owed salary so we don't need to update
+            // their last payroll date
+            // No need to use SafeMath as we already know _paymentAmount <= accruedSalary
             employee.accruedSalary = accruedSalary - _paymentAmount;
-        } else if (accruedSalary > 0) {
-            // employee is cashing out a mixed amount between previous and current owed salaries,
-            // then we need to set the accrued salary to zero and update the last payroll date
-            // there's no need to use safemath here since we already know that _paymentAmount is > accruedSalary
+            return;
+        }
+
+        // Employee is cashing out some of their currently owed salary so their last payroll date
+        // needs to be modified based on the amount of salary paid
+        uint256 currentSalaryPaid = _paymentAmount;
+        if (accruedSalary > 0) {
+            // Employee is cashing out a mixed amount between previous and current owed salaries;
+            // first use up their accrued salary
             employee.accruedSalary = uint256(0);
-            uint256 remainder = _paymentAmount - accruedSalary;
-            employee.lastPayroll = (remainder == _currentOwedSalary) ? getTimestamp64() : _getLastPayroll(_employeeId, remainder);
-        } else {
-            // employee is only cashing out some current owed salary, and there is no previous owed salary,
-            // then we only need to update the last payroll date
-            employee.lastPayroll = (_paymentAmount == _currentOwedSalary) ? getTimestamp64() : _getLastPayroll(_employeeId, _paymentAmount);
+            // No need to use SafeMath here as we already know _paymentAmount > accruedSalary
+            currentSalaryPaid = _paymentAmount - accruedSalary;
         }
+        employee.lastPayroll = _getLastPayrollDate(_employeeId, currentSalaryPaid);
     }
```

Comment:
I found this reorganizing of logic a bit more clear than the previous version, since in the last two cases we want to update the employee's `lastPayroll`.